registry/auth/htpasswd/access.go

```diff
@@ -9,13 +9,17 @@ import (
 	"fmt"
 	"net/http"
 	"os"
+	"sync"
 
 	"github.com/docker/distribution/context"
 	"github.com/docker/distribution/registry/auth"
 )
 
 type accessController struct {
 	realm    string
+	path     string
+	modtime  string
+	mtx      *sync.Mutex
```

Comment:
I wonder whether the mutex have the effect on locking those critical regions, if a mutex is initialized in on each parallel request instead of being defined uniquely as a global shared variable?